enumerables.rb

```diff
@@ -0,0 +1,112 @@
+# Creating the enumerable module
+module Enumerables 
+    #my_each method
+    def my_each 
+        i = 0
+        while i < self.length
+            yield(self[i])
+            i+= 1
+        end
+    end
+      
+#my_each_with_index method
+def my_each_with_index
+    index = 0
+    i = 0
+    while i < self.length
+    yield(self[i], index)
+    index += 1
+    i+= 1
+    end
+end
+
+#my_select method
+def my_select
+arr = []
+self.my_each {|el| arr << el if yield(el)}
+arr 
+end
+
+#my_all? method
+def my_all? 
+    result= false
+    self.my_each do |el| 
+    if yield(el) 
+    result = true
+    else 
+     false
+    end
+    result
+    end
+
+end
+
+#my_any?
+
+def my_any?
+
+    self.my_each {|el| return true if yield(el)}
+    false    
+end
+
+
+def my_none?
+    result = true
+    self.my_each do |el|
+        if yield(el)
+        return false
+        end
+
+    end
+  result
+end 
+
+#count method
+def my_count(el)
+    return self.length if el == nil
+    return self.my_select(el) if el
```

Comment:
Please remember that `my_select` does not expect an argument to be passed, so an error would be thrown. It does, however, expect a block.

Reply:
Also, please remember that `my_select` returns an array, so maybe you'd need to return the length in this case

Reply:
It's really cool that you decided to use my_select though! Was really smart move! :) Shows understanding of what `my_select` works.